castai/resource_castai_cluster.go

```diff
@@ -217,7 +402,107 @@ func waitForClusterToReachStatusFunc(ctx context.Context, client *sdk.ClientWith
 				return resource.RetryableError(fmt.Errorf("waiting for cluster to reach %q status, id=%q name=%q, status=%s", targetStatus, cluster.Id, cluster.Name, cluster.Status))
 			}
 		}
-
 		return resource.NonRetryableError(fmt.Errorf("cluster has reached unexpected status, id=%q name=%q, status=%s", cluster.Id, cluster.Name, cluster.Status))
 	}
 }
+
+func expandCPUlimitPolicy(params interface{}) sdk.ClusterLimitsCpu {
+	var clusterLimitsCPU sdk.ClusterLimitsCpu
+	for _, val := range params.([]interface{}) {
+		cpuData := val.(map[string]interface{})
+		clusterLimitsCPU = sdk.ClusterLimitsCpu{
+			MaxCores: int64(cpuData[PolicyFieldClusterLimitsCPUmax].(int)),
+			MinCores: int64(cpuData[PolicyFieldClusterLimitsCPUmin].(int)),
+		}
+	}
+	return clusterLimitsCPU
+}
+
+func expandNodeDownscalerPolicy(params interface{}) *sdk.NodeDownscaler {
+
+	var nodeDownscalerPolicy sdk.NodeDownscaler
+	var nodeDownscalerPolicyEnabled sdk.NodeDownscalerEmptyNodes
+
+	for _, val := range params.([]interface{}) {
+		ndData := val.(map[string]interface{})
+		nodeDownscalerEnabled := ndData[PolicyFieldEnabled].(bool)
+		nodeDownscalerPolicyEnabled = sdk.NodeDownscalerEmptyNodes{
+			Enabled: &nodeDownscalerEnabled,
+		}
+	}
+
+	nodeDownscalerPolicy = sdk.NodeDownscaler{
```

Comment:
@ernestas2k now it's harder for me to read, extracting sdk.NodeDownscaler to expander. Do you have other ideas how to make pretty?

Reply:
My original though was to not have many `func expand*` functions but have one `expandAutoscalerPolicies(...)`. Your thoughts?